### ISSUE TITLE
Regular stub installer does not send download-complete and installer-run pings

### ISSUE THREAD

**Issue by btlechowski** [brave/brave-browser#2917]
## Steps to Reproduce

   1. Uninstall Brave though `Control Panel`
   2. Delete `Download` and `Install` from `c:\Program Files (x86)\BraveSoftware\Update\`
   3. Prepare Fiddler as described in https://github.com/brave/brave-browser/issues/2784#issuecomment-453298218
4. Run regular stub installer (BraveBrowserSetup.exe)

## Actual result:

Only `startup` ping is sent
Pings `download-complete` and `installer-run` are not sent
![image](https://user-images.githubusercontent.com/34715963/51006598-4c164c80-1544-11e9-850a-d2a0197bd34d.png)

## Expected result:
All 3 pings are sent

## Reproduces how often: 

always

## Brave version (brave://version info)

Brave | 0.58.20 Chromium: 71.0.3578.98 (Official Build) (64-bit)
-- | --
Revision | 15234034d19b85dcd9a03b164ae89d04145d8368-refs/branch-heads/3578@{#897}
OS | Windows 7

Also reproduced on Windows 10 x64

cc @emerick @brave/legacy_qa

**Comment by emerick** [comment]
@btlechowski I would check if @aubrey can possibly pull the server logs for those requests, it could just be a hiccup in Fiddler. I only ever ran and tested with the regular (non-silent) installer, so it would be weird if it wasn't working (and I even downloaded the release candidate regular stub installer today and confirmed that it sent all three pings).

**Comment by emerick** [comment]
@simonhong I'll investigate in the morning, but any thoughts why there could be a difference here between the regular stub installer and the silent one?

**Comment by emerick** [comment]
Spoke with @simonhong and it seems to occur when performing an admin install. When you receive the UAE prompt, if you choose "Yes" it will do an admin install and if you choose "No" it will do a non-admin install. It seems that the admin install leads to this problem.

**Comment by emerick** [comment]
Upon further investigation, it seems to be related to trying to read the app guid out of the registry when running in admin mode; the key isn't there. We don't need to store the app_guid in the registry, since we have access to the app_guid in the app bundle. So we can fix this problem. But, we also read the referral code from the registry and there is no clear way around that.

@bbondy Just a heads up on this one.

**Comment by bbondy** [comment]
let's not block

**Comment by emerick** [comment]
We won't consult the registry at all in order to avoid this problem. For the app guid, we'll parse it out of the tag in the metainstaller and we'll access it via the app pointer. For the referral code, we'll parse it out of our filename as always and then append it to the tag that we forward to the installer (again accessing it via the app pointer, since Omaha already supports the concept of a referral id).

**Comment by emerick** [comment]
@btlechowski I discovered an easier way to debug this, when the time comes. If you create a file named `c:\BraveUpdate.ini` with the correct settings (may need to use admin privileges to write to that location), Omaha will log all relevant information to `C:\ProgramData\BraveSoftware\Update\Log\BraveUpdate.log` including the requests and responses associated with the Brave server. I'll attach the .ini file that I used for testing, although note that I had to zip it to add it here.

[BraveUpdate.zip](https://github.com/brave/brave-browser/files/2756022/BraveUpdate.zip)
